services/packagist/packagist-base.js

```diff
@@ -6,9 +6,7 @@ const packageSchema = Joi.array().items(
   Joi.object({
     version: Joi.string().required(),
     require: Joi.alternatives(
-      Joi.object({
-        php: Joi.string(),
-      }).required(),
+      Joi.object().pattern(Joi.string(), Joi.string()).required(),
```

Comment:
This change was needed so that we accept different dependencies' names for the package from the response,
for example:
```js
  "require": {
          "php": ">=8.1",
          "composer-runtime-api": ">=2.1",
          "ext-xml": "*",
          "friendsofphp/proxy-manager-lts": "^1.0.2",
          "doctrine/event-manager": "~1.0",
          "doctrine/persistence": "^2|^3",
          "twig/twig": "^2.13|^3.0.4",
          "psr/cache": "^2.0|^3.0",
          "psr/container": "^1.1|^2.0",
          "psr/event-dispatcher": "^1.0",
          "psr/link": "^1.1|^2.0",
          "psr/log": "^1|^2|^3",
          "symfony/contracts": "^2.5|^3.0",
          "symfony/polyfill-ctype": "~1.8",
          "symfony/polyfill-intl-grapheme": "~1.0",
          "symfony/polyfill-intl-icu": "~1.0",
          "symfony/polyfill-intl-idn": "^1.10",
          "symfony/polyfill-intl-normalizer": "~1.0",
          "symfony/polyfill-mbstring": "~1.0",
          "symfony/polyfill-uuid": "^1.15"
        },
```
from https://packagist.org/packages/symfony/symfony.json

Reply:
Yep - this is fine. As long as we're throwing an exception if the dependency we're looking for isn't found in `require` loosening the validation here is fine.